src/logic/safe/transactions/offchainSigner/index.js

```diff
@@ -0,0 +1,28 @@
+// @flow
+
+import { getEIP712Signer } from './EIP712Signer'
+import { generateEthSignature } from './ethSigner'
+
+// 1. we try to sign via EIP-712 if user's wallet supports it
+// 2. If not, try to use eth_sign (Safe version has to be >1.1.1)
+// If eth_sign, doesn't work continue with the regular flow (on-chain signatures, more in createTransaction.js)
+
+const signingFuncs = [getEIP712Signer('v3'), getEIP712Signer(), generateEthSignature]
+
+export const tryOffchainSigning = async txArgs => {
+  let signature
+  for (let signingFunc of signingFuncs) {
+    try {
+      signature = await signingFunc(txArgs)
+
+      break
+    } catch (err) {
+      console.error(err)
+      if (err.code === 4001) {
```

Comment:
Was this a trial error thing to find the error code, or is this documented somewhere (e.g. do all wallets use the same error code)

Reply:
This is metamask's error code when users rejects signature request. Added a comment